### PR TITLE
fix(getObjectValueByPath): fix getObjectValueByPath

### PR DESCRIPTION
**Pull request by nekosaur**
## Description
Return early if key exists directly on object to improve performance

## Motivation and Context
solves #7302 

## How Has This Been Tested?
playground, unit test

## Markup:

<details>

```vue
<template>
  <v-container>
    <v-text-field v-model="search" label="search"></v-text-field>
    <v-data-table
        :headers="headers"
        :items="moreDesserts"
        :search="search"
      ></v-data-table>
  </v-container>
</template>

<script>
  export default {
    data () {
      return {
        search: '',
        headers: [
          {
            text: 'Dessert (100g serving)',
            align: 'left',
            sortable: false,
            value: 'name'
          },
          { text: 'Calories', value: 'calories' },
          { text: 'Fat (g)', value: 'fat' },
          { text: 'Carbs (g)', value: 'carbs' },
          { text: 'Protein (g)', value: 'protein' },
          { text: 'Iron (%)', value: 'iron' }
        ],
        desserts: [
          {
            value: false,
            name: 'Frozen Yogurt',
            calories: 159,
            fat: 6.0,
            carbs: 24,
            protein: 4.0,
            iron: '1%'
          },
          {
            value: false,
            name: 'Ice cream sandwich',
            calories: 237,
            fat: 9.0,
            carbs: 37,
            protein: 4.3,
            iron: '1%'
          },
          {
            value: false,
            name: 'Eclair',
            calories: 262,
            fat: 16.0,
            carbs: 23,
            protein: 6.0,
            iron: '7%'
          },
          {
            value: false,
            name: 'Cupcake',
            calories: 305,
            fat: 3.7,
            carbs: 67,
            protein: 4.3,
            iron: '8%'
          },
          {
            value: false,
            name: 'Gingerbread',
            calories: 356,
            fat: 16.0,
            carbs: 49,
            protein: 3.9,
            iron: '16%'
          },
          {
            value: false,
            name: 'Jelly bean',
            calories: 375,
            fat: 0.0,
            carbs: 94,
            protein: 0.0,
            iron: '0%'
          },
          {
            value: false,
            name: 'Lollipop',
            calories: 392,
            fat: 0.2,
            carbs: 98,
            protein: 0,
            iron: '2%'
          },
          {
            value: false,
            name: 'Honeycomb',
            calories: 408,
            fat: 3.2,
            carbs: 87,
            protein: 6.5,
            iron: '45%'
          },
          {
            value: false,
            name: 'Donut',
            calories: 452,
            fat: 25.0,
            carbs: 51,
            protein: 4.9,
            iron: '22%'
          },
          {
            value: false,
            name: 'KitKat',
            calories: 518,
            fat: 26.0,
            carbs: 65,
            protein: 7,
            iron: '6%'
          }
        ],
        count:0,
      }
    },
    watch: {
      moreDesserts(newList){
        this.count = newList.length
      }
    },
    computed: {
      moreDesserts() {
        let all = []
        for (let i = 0; i < 20000; i++) {
          this.desserts.forEach((c) => {
            all.push({ ...c, name: `${c.name}_${i}` })
          })
        }
        return all
      }
    }
  }
</script>

```
</details>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/src/util/helpers.ts [hunk 1]
```diff
@@ -214,9 +214,10 @@ export function deepEqual (a: any, b: any): boolean {
   return props.every(p => deepEqual(a[p], b[p]))
 }
 
-export function getObjectValueByPath (obj: object, path: string, fallback?: any): any {
+export function getObjectValueByPath (obj: any, path: string, fallback?: any): any {
```
Comment (KaelWD) [inline packages/vuetify/src/util/helpers.ts]: `unknown`?

File: packages/vuetify/src/util/helpers.ts [hunk 2]
```diff
@@ -214,9 +214,10 @@ export function deepEqual (a: any, b: any): boolean {
...
   // credit: http://stackoverflow.com/questions/6491463/accessing-nested-javascript-objects-with-string-key#comment55278413_6491621
-  if (!path || path.constructor !== String) return fallback
+  if (obj == null || !path || path.constructor !== String) return fallback
```
Comment (KaelWD) [inline packages/vuetify/src/util/helpers.ts]: `typeof path`?

File: packages/vuetify/src/util/helpers.ts [hunk 3]
```diff
@@ -214,9 +214,10 @@ export function deepEqual (a: any, b: any): boolean {
...
+  if (obj[path] !== undefined) return obj[path]
```
Comment (KaelWD) [inline packages/vuetify/src/util/helpers.ts]: `path in object`? depends what you want the behaviour to be for values that exist but are undefined

File: packages/vuetify/src/util/helpers.ts (hunk 2 above)
Comment (sh7dm) [reply packages/vuetify/src/util/helpers.ts]: :+1: https://stackoverflow.com/questions/39802506/constructor-vs-typeof-to-detect-type-in-javascript

File: packages/vuetify/src/util/helpers.ts (hunk 3 above)
Comment (nekosaur) [reply packages/vuetify/src/util/helpers.ts]: imo keys with undefined as value should be treated as key not existing

**Pull request merged**